options.go

```diff
@@ -106,14 +106,33 @@ func WithHostPortAccess(ports ...int) CustomizeRequestOption {
 	}
 }
 
+// WithName will set the name of the container.
+func WithName(containerName string) CustomizeRequestOption {
+	return func(req *GenericContainerRequest) error {
+		if containerName == "" {
+			return errors.New("container name must be provided")
+		}
+		req.Name = containerName
+		return nil
+	}
+}
+
+// WithNoStart will prevent the container from being started after creation.
+func WithNoStart() CustomizeRequestOption {
+	return func(req *GenericContainerRequest) error {
+		req.Started = false
+		return nil
+	}
+}
```

Comment:
suggestion: pass in start so it can he used to set and unset started?

Reply:
I considered it,  although I see the Started state as the default, so I won't Don't want users to explicitly set it

Reply:
Unfortunately while the intention might be for started to be the default, as this is a bool and the default value is false it actually isn't so it's totally possible to need to set it. If it was NoStart allowing the zero value this would make more sense. Another example of why keeping fields private provides implementation flexibility.

Not a blocker but something to consider

Reply:
> Another example of why keeping fields private provides implementation flexibility.

Indeed!

Unfortunately, the request struct was here since the beginning. But with all the functional options, in combination with the Run function, we are really close to start deprecating the usage of the request struct in favour of the Run function + options.

I'd keep it as is, so that users don't have to worry about setting true/false: the Run functions (in modules and core) will always _run_ the container, so the default will be true.